File: samples/src/main/java/tk/zielony/carbonsamples/SampleActivityItem.java
package tk.zielony.carbonsamples;

import android.app.Activity;

import java.io.Serializable;

public class SampleActivityItem implements Serializable {
    private Class<? extends Activity> activityClass;
    private String name;
    private boolean beta = false;

    public SampleActivityItem(Class<? extends Activity> activityClass, String name) {
        this.activityClass = activityClass;
        this.name = name;
    }

    public SampleActivityItem(Class<? extends Activity> activityClass, String name, boolean beta) {
        this.activityClass = activityClass;
        this.name = name;
        this.beta = beta;
    }

    public Class<? extends Activity> getActivityClass() {
        return activityClass;
    }

    public String getName() {
        return name;
    }

    public boolean isBeta() {
        return beta;
    }
}

File: samples/src/main/java/tk/zielony/carbonsamples/SampleActivityItem.kt
package tk.zielony.carbonsamples

import android.app.Activity
import java.io.Serializable

open class SampleActivityGroup : Serializable {
    var activityClass: Class<out Activity>? = null
        private set
    var name: Int? = null
        private set
    var icon = 0
        private set

    @JvmOverloads
    constructor(activityClass: Class<out Activity>) : super(
    ) {
        this.activityClass = activityClass
        activityClass.getAnnotation(SampleAnnotation::class.java)?.let {
            this.name = it.titleId
            this.icon = it.iconId
        }
    }
}

class SampleActivityItem : SampleActivityGroup {
    var isBeta = false
        private set
    var isStarred: Boolean = false

    @JvmOverloads
    constructor(activityClass: Class<out Activity>, beta: Boolean = false) : super(activityClass) {
        this.isBeta = beta
    }
}
